core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java

```diff
@@ -237,13 +239,21 @@ protected void addLocations(RebindContext rebindContext, EntityMemento memento)
             }
         }
     }
-    
+
+    @Override
     protected void instanceRebind(AbstractBrooklynObject instance) {
         Preconditions.checkState(instance == entity, "Expected %s and %s to match, but different objects", instance, entity);
         Lifecycle expectedState = ServiceStateLogic.getExpectedState(entity);
         if (expectedState == Lifecycle.STARTING || expectedState == Lifecycle.STOPPING) {
-            LOG.warn("Entity "+entity);
+            LOG.warn("Entity {} goes on-fire because it was in state {} on rebind",
+                    entity, ServiceStateLogic.getExpectedState(entity));
+            LOG.warn("not-up-indicators={}", entity.getAttribute(Attributes.SERVICE_NOT_UP_INDICATORS));
             ServiceStateLogic.setExpectedState(entity, Lifecycle.ON_FIRE);
+
+            if (!(entity instanceof BasicApplication) && entity.getLocations().isEmpty()) {
+                LOG.warn("Rebind happened during provisioning for entity {} with parent {}" +
```

Comment:
Unfortunately this is too general. There are some entity types that will not necessarily have locations (e.g. that are not software processes, but instead bind to an API etc. For example, the `GistGenerator` docs example in http://docs.cloudsoft.io/blueprints/java/defining-and-deploying.html.

Let's take this out, and come back to it in a separate PR.